docs/dev/docker.rst

```diff
@@ -0,0 +1,51 @@
+.. _docker:
+
+Builds and Dependency Management
+================================
+
+General structure of Programming Exercise Execution
+---------------------------------------------------
+
+Artemis uses Docker containers for running programming exercises. This ensures that the code of the students doesn't have direct access to the hardware of the build agents.
+To reduce the needed time of each test run, these docker containers already contain often used dependencies like JUnit.
```

Comment:
Suggestion for small linguistic changes: "Artemis uses Docker containers to run programming exercises. This ensures that the students' code does not have direct access to the build agents' hardware.
To reduce the time required for each test run, these Docker containers already include commonly used dependencies such as JUnit."